legacy/src/app/controllers/domain_details.js

```diff
@@ -129,7 +129,7 @@ function DomainDetailsController(
 
   // Called to check if the space can be deleted.
   $scope.canBeDeleted = function () {
-    if (angular.isObject($scope.domain)) {
+    if (angular.isObject($scope.domain) && $scope.domain.rrsets) {
```

Comment:
This could just be `if ($scope.domain?.rrsets) {`

Reply:
Also, is it possible this should remain unchanged and the line below should be: `return !$scope.domain.rrsets || $scope.domain.rrsets.length === 0;`?

Reply:
Sure, I would have used optional chaining in modern code but I didn't want to introduce a different way of coding in the middle of a file. This video is to be replaced in the near future. 

The second suggestion seems very complex to understand for the next developer that comes a long. I wanted to make it simple to read.